pkg/apis/build/v1alpha2/build_pod_test.go

```diff
@@ -670,22 +811,46 @@ func testBuildPod(t *testing.T, when spec.G, it spec.S) {
 			pod, err := build.BuildPod(config, buildContext)
 			require.NoError(t, err)
 
-			assert.Equal(t, pod.Spec.InitContainers[1].Name, "detect")
-			assert.Contains(t, pod.Spec.InitContainers[1].Env, corev1.EnvVar{Name: "CNB_PLATFORM_API", Value: "0.6"})
-			assert.Equal(t, pod.Spec.InitContainers[1].Image, builderImage)
+			assert.Equal(t, pod.Spec.InitContainers[2].Name, "detect")
+			assert.Contains(t, pod.Spec.InitContainers[2].Env, corev1.EnvVar{Name: "CNB_PLATFORM_API", Value: "0.8"})
+			assert.Equal(t, pod.Spec.InitContainers[2].Image, builderImage)
 			assert.Equal(t, []string{
 				"layers-dir",
 				"platform-dir",
 				"workspace-dir",
 				"binding-metadata-some-v1alpha1-binding",
 				"binding-secret-some-v1alpha1-binding",
-			}, names(pod.Spec.InitContainers[1].VolumeMounts))
+			}, names(pod.Spec.InitContainers[2].VolumeMounts))
 		})
 
 		it("configures analyze step", func() {
 			pod, err := build.BuildPod(config, buildContext)
 			require.NoError(t, err)
 
+			assert.Equal(t, pod.Spec.InitContainers[1].Name, "analyze")
+			assert.Contains(t, pod.Spec.InitContainers[1].Env, corev1.EnvVar{Name: "CNB_PLATFORM_API", Value: "0.8"})
+			assert.Equal(t, pod.Spec.InitContainers[1].Image, builderImage)
+			assert.Equal(t, []string{
+				"layers-dir",
+				"workspace-dir",
+				"home-dir",
+				"cache-dir",
```

Comment:
Technically it is not necessary to mount in the cache directory, because the analyzer (for newer platform api) will not use it.

Reply:
Yup - I just left it there since it made the code simple and it didn't do any harm. But can move it out if needed.

Reply:
@natalieparellano removed it conditionally based on platform api.